File: backend/app/schemas/document.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.common import Classification

DocumentCategory = Literal[
    "bieu_mau",
    "huong_dan",
    "quy_che_quy_dinh",
    "ke_hoach",
    "bao_cao",
    "van_ban_chi_dao",
]


class DocumentMetaUpdate(BaseModel):
    title: str = Field(..., max_length=255)
    description: Optional[str] = None
    category: DocumentCategory
    classification: Classification = "noi_bo"


class DocumentOut(BaseModel):
    id: int
    title: str
    description: Optional[str]
    category: DocumentCategory
    file_url: str
    file_name: str
    file_size: int
    content_type: str
    classification: Classification
    uploaded_by_id: int
    uploaded_by_full_name: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/document.ts
import type { Classification } from './common'

export type DocumentCategory =
  | 'bieu_mau'
  | 'huong_dan'
  | 'quy_che_quy_dinh'
  | 'ke_hoach'
  | 'bao_cao'
  | 'van_ban_chi_dao'

export const DOCUMENT_CATEGORY_LABELS: Record<DocumentCategory, string> = {
  bieu_mau: 'Biểu mẫu',
  huong_dan: 'Hướng dẫn',
  quy_che_quy_dinh: 'Quy chế – Quy định',
  ke_hoach: 'Kế hoạch',
  bao_cao: 'Báo cáo',
  van_ban_chi_dao: 'Văn bản chỉ đạo',
}

export interface DocumentFormData {
  title: string
  category: DocumentCategory
  description: string
  classification: Classification
}

export interface DocumentItem {
  id: number
  title: string
  description: string | null
  category: DocumentCategory
  file_url: string
  file_name: string
  file_size: number
  content_type: string
  classification: Classification
  uploaded_by_id: number
  uploaded_by_full_name: string
  created_at: string
}
